From: Mohamad Al-Jaf <mohamadaljaf@gmail.com>
Subject: [PATCH v11 8/8] wimgapi: Add WIMSetFileIOCallbackTemporaryPath stub.
Message-Id: <20220412030134.405685-5-mohamadaljaf@gmail.com>
Date: Mon, 11 Apr 2022 23:01:34 -0400
In-Reply-To: <20220412030134.405685-1-mohamadaljaf@gmail.com>
References: <20220412030134.405685-1-mohamadaljaf@gmail.com>

The Windows Media Creation Tool crashes on this
unimplemented function.

Signed-off-by: Mohamad Al-Jaf <mohamadaljaf@gmail.com>
---
 dlls/wimgapi/main.c       | 7 +++++++
 dlls/wimgapi/wimgapi.spec | 2 +-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/dlls/wimgapi/main.c b/dlls/wimgapi/main.c
index 11111111111..11111111111 100644
--- a/dlls/wimgapi/main.c
+++ b/dlls/wimgapi/main.c
@@ -54,3 +54,10 @@ BOOL WINAPI WIMInitFileIOCallbacks(LPVOID callbacks)
     SetLastError(ERROR_CALL_NOT_IMPLEMENTED);
     return 0;
 }
+
+BOOL WINAPI WIMSetFileIOCallbackTemporaryPath(LPWSTR path)
+{
+    FIXME("(%s) stub\n", debugstr_w(path));
+    SetLastError(ERROR_CALL_NOT_IMPLEMENTED);
+    return 0;
+}
diff --git a/dlls/wimgapi/wimgapi.spec b/dlls/wimgapi/wimgapi.spec
index 11111111111..11111111111 100644
--- a/dlls/wimgapi/wimgapi.spec
+++ b/dlls/wimgapi/wimgapi.spec
@@ -31,7 +31,7 @@
 @ stdcall WIMRegisterMessageCallback(long ptr ptr)
 @ stub WIMRemountImage
 @ stub WIMSetBootImage
-@ stub WIMSetFileIOCallbackTemporaryPath
+@ stdcall WIMSetFileIOCallbackTemporaryPath(wstr)
 @ stub WIMSetImageInformation
 @ stub WIMSetReferenceFile
 @ stub WIMSetTemporaryFile

-- 
2.39.0

